src/components/core/geoChart/geoChartColor.ts

```diff
@@ -89,18 +90,18 @@ export function getColorPaletteMapping(segmentItems: string[] = []): IObjectMapp
  *           border: "rgb(127,224,198)",
  *           background: "rgb(215,242,250)",
  *      }]
- * @param colors
+ * @param colorValues
  * @param segmentByItems
  */
 export function getPushpinColors(
-    colors: number[],
+    colorValues: number[],
     segmentByItems: Execution.IResultHeaderItem[] = [],
 ): IPushpinColor[] {
-    if (!colors.length) {
+    if (!colorValues.length) {
         return [];
     }
 
-    const colorsWithoutNull = compact(colors);
+    const colorsWithoutNull = colorValues.filter((color: number) => !(color === null || color === undefined));
```

Comment:
should we care about empty string and Nan?

Reply:
No, we shouldn't. We already convert them to number in `transformPushpinDataSource`
and check `isNaN` in 40th line in same file